operations/observability/Observability.md

```diff
@@ -0,0 +1,300 @@
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=grafanaStorageLogger t=2023-02-08T23:06:38.54+0000 lvl=info msg="storage starting"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert t=2023-02-08T23:06:38.54+0000 lvl=info msg="warming cache for startup"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=ngalert.multiorg.alertmanager t=2023-02-08T23:06:38.54+0000 lvl=info msg="starting MultiOrg Alertmanager"
+Feb 08 23:06:38 redbull-pit grafana[82549]: logger=http.server t=2023-02-08T23:06:38.55+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http>
+```
+
+## IUF Timing dashboard
+
+### Timing dashboard features
+
+- Use Argo workflows to collect the install/upgrade timing details.
+- Create Argo workflows to get metrics from the details available.
+- Generate Prometheus install/upgrade timing metrics.
+- Create Grafana dashboard using Prometheus metrics.
+
+### Prometheus metrics using Argo Workflow
+
+Able to get the start-time and end-time as Prometheus labels for the operations using Argo metrics approach. Added record time stamp task at the beginning and end of the operation template using timestamp output parameter as an input to the metrics.
+
+Metrics captured for the operations:
+
+- start time
+- end time
+- duration
+- status
+- product name
+- product version
+
+Metrics captured for the stage:
+
+- stage name
+- stage type
+- stage start time (derived from min(start time of the all the operations presented in the stage)
+- stage end time (derived from max(start time of the all the operations presented in the stage)
+- stage duration (stage end time - stage start time)
+- stage status (derived from all the operations status)
+
+Metrics captured for the product:
+
+- product name
+- product start time (derived from the start time of the process-media stage)
+- product end time (derived from the start time of the post-install-check stage)
+- product status (derived from the status of the stage)
+
+### Timing dashboard
+
+- Created dynamic top-down and bottom-up dashboard to track install/upgrade status of any product, stage and operation.
+- Dashboard calculates the execution time for the install/ upgrade of any product, stage and operation.
+- The status of the install/ upgrade for product, stage and operation are Failed/Succeeded.
+- There is dropdown for selection of the product, stage and operation. By default all are selected.
+- Separate sections created in dashboard to see the details of product, stage and operation.
+- Graph showing duration for each stage & operation is added.
+
+![IUF timing dashboard](../../img/operations/TimingDashboard.png "Timing Dashboard")
+
+## Goss tests for PIT and NCN
+
+### Goss test overview
+
+Goss test logs are scraped using Grok-exporter and visualization using captured data. CSM-testing repository has Goss-testing with all tests, suites and scripts to execute these Goss tests.
```

Comment:
If by CSM-testing repository you mean the github `csm-testing` repository, then it should be changed to match that. Although in that case, I would also recommend linking to the repository.

Reply:
Made the change and added the link to csm-testing repository.